otherlibs/unix/read_unix.c

```diff
@@ -53,3 +53,14 @@ CAMLprim value caml_unix_read_bigarray(value fd, value vbuf,
   if (ret == -1) caml_uerror("read_bigarray", Nothing);
   CAMLreturn(Val_long(ret));
 }
+
+intnat caml_unix_fast_read(value fd, value buf, intnat ofs, intnat len)
+{
+  return read(Int_val(fd), &Byte(buf, ofs), len);
+}
+
+CAMLprim value caml_byte_unix_fast_read(value fd, value buf, value vofs,
+                                   value vlen) {
+  CAMLparam0();
+  CAMLreturn(Int_val(caml_unix_fast_read(fd,buf,Int_val(vofs),Int_val(vlen))));
```

Comment:
```suggestion
  CAMLreturn(Val_int(caml_unix_fast_read(fd,buf,Int_val(vofs),Int_val(vlen))));
```
I believe you wanted to call the macro `Val_int` here.